### PR TITLE
Optimize listeners

### PR DESCRIPTION
**Pull request by viktorprogger**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | ❌

### CODE REVIEW

File: src/EventConfigurator.php [hunk 1]
```diff
@@ -66,15 +67,14 @@ public function registerListeners(array $eventListeners): void
...
+                    if (is_array($callable) && !is_object($callable[0])) {
+                        $callable = [$container->get($callable[0]), $callable[1]];
+                    }
 
-                $this->listenerProvider
-                    ->attach(
-                        static fn($event) => $injector->invoke($callable, [$event]),
-                        $eventName
-                    );
+                    $injector->invoke($callable, [$event]);
```
Comment (yiiliveext) [inline src/EventConfigurator.php:75]: Where did you lose return:)

Comment (viktorprogger) [reply src/EventConfigurator.php:75]: thnx

File: src/EventConfigurator.php [hunk 2]
```diff
@@ -66,15 +67,14 @@ public function registerListeners(array $eventListeners): void
                     );
                 }
 
-                if (is_array($callable) && !is_object($callable[0])) {
-                    $callable = [$this->container->get($callable[0]), $callable[1]];
-                }
+                $listener = static function ($event) use ($injector, $callable, $container) {
```
Comment (roxblnfk) [inline src/EventConfigurator.php:70]:
```suggestion
                $listener = static function (object $event) use ($injector, $callable, $container) {
```

File: src/EventConfigurator.php [hunk 3]
```diff
@@ -33,6 +33,7 @@ public function __construct(Provider $listenerProvider, ContainerInterface $cont
     public function registerListeners(array $eventListeners): void
     {
         $injector = new Injector($this->container);
+        $container = $this->container;
```
Comment (roxblnfk) [inline src/EventConfigurator.php:36]:
```suggestion
        $container = $this->container;
        $injector = new Injector($container);
```

**Pull request merged**